samples/nodejs-compat-module/foo.js

```diff
@@ -0,0 +1,10 @@
+process.nextTick(() => {
+  console.log('this should work');
+});
+
+module.exports = {
+  // Buffer is a global...
+  Buffer,
+  // require can be called synchronously inline.
+  util: require('util'),
+}
```

Comment:
The idea here is that we would have the ability to take a module that is written for Node.js and have a greater chance of it Just Working without modifications since the expected Node.js globals would be there (e.g. `Buffer`, `process.nextTick`, etc) and things like `require('util')` without the `node:` specifier-prefix would do the right thing.

It is still not a 100% guarantee since there is so much of the Node.js API surface that we are not implementing.